### ISSUE TITLE
Parameter "is not accessed" warning when overloading functions.

### ISSUE THREAD

**Issue by mnicstruwig** [microsoft/pyright#1863]
**Describe the bug**
When typehinting with function overloads, the parameters to the function are labelled as "not accessed" by `pyright`. 

**To Reproduce**
Example that reproduces the problem:
```python
from typing import overload

@overload
def my_func(x: int) -> int:          ~~~ "x" is not accessed
    ...

@overload
def my_func(x: float) -> float:     ~~~ "x" is not accessed
    ...

def my_func(x: Union[int, float]) -> Union[int, float]:
    return x + x
```

**Expected behavior**
No warning for unaccessed parameters to be raised for overloaded signatures.

**VS Code extension or command-line**
`lsp-mode` extension on Emacs.
Pyright version: `1.1.139`

Please let me know if I've otherwise forgotten something (I'm new to signature overloads).
Thanks!

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by mnicstruwig** [comment]
Great, thanks!

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.140, which I just published. It will also be in the next release of pylance.
